ase/scripts/generate_ase_environment.py

```diff
@@ -282,6 +288,130 @@ def config_sources(fd, filelist):
     return json_file
 
 
+#
+# Qsys has a compilation step.  When using case #1 above, detect Qsys
+# sources, compile them, and include the generated Verilog in the
+# simulation.
+#
+def config_qsys_sources(filelist, vlog_srcs):
+    # Get all the sources.  rtl_src_config will emit all relevant source
+    # files, one per line.
+    try:
+        srcs = commands_list_getoutput(
+            "rtl_src_config --qsys --abs".split(" ") + [filelist])
+    except Exception:
+        errorExit("failed to read sources from {0}".format(filelist))
+
+    # Collect two sets: qsys source files and the names of directories into
+    # which generated Verilog will be written.  The directory names match
+    # the source names.
+    qsys_srcs = []
+    ip_dirs = []
+    srcs = srcs.split('\n')
+    for s in srcs:
+        if (s):
+            # Record all build target directories
+            ip_dirs.append(os.path.splitext(s)[0])
+
+            # Collect all qsys files
+            if (s.lower()[-5:] == '.qsys'):
+                qsys_srcs.append(s)
+
+    # Any Qsys files found?
+    if (not qsys_srcs):
+        return None
+
+    # First step: copy the trees holding Qsys sources to a temporary tree
```

Comment:
Does the temporary tree get deleted after simulation? Also, make sure the temporary tree is cleaned up on error.

Reply:
The temporary tree is inside the directory created for simulation, just like the old version. We don't currently clean up the created simulation tree on any error. I suppose we could, but that could be done at a higher level than this change since this source of errors is only one of many and the action would be the same.

Reply:
Ok.